### PR TITLE
Hashicorp Vault documentation

### PR DESCRIPTION
**Pull request by zivkovicmilos**
# Description

This PR adds usage documentation on the [Hashicorp Vault support PR](https://github.com/0xPolygon/polygon-sdk/pull/209).

### CODE REVIEW

File: docs/how-tos/howto-setup-hashicorp-vault.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+## Step 1 - Generate the secrets manager configuration
+
+In order for the Polygon SDK to be able to seamlessly communicate with the Vault server, it needs to parse an already
+generated config file, which contains all the necessary information for secret storage on Vault.
+
+To generate the configuration, run the following command:
+
+```bash
+go run main.go secrets-manager generate --dir <PATH> --token <TOKEN> --server-url <SERVER_URL> --name <NODE_NAME>
```
Comment (Kourin1996) [inline docs/how-tos/howto-setup-hashicorp-vault.md:50]: This is correct one?
```
go run main.go secrets generate --dir <PATH> --token <TOKEN> --server-url <SERVER_URL> --name <NODE_NAME>
```

File: docs/how-tos/howto-setup-hashicorp-vault.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+Secrets are stored on the following base path: `secrets/node_name`
+:::
+
+## Step 2 - Initialize IBFT using the configuration
+
+Now that the configuration file is present, we can initialize required IBFT data (secret keys) with the configuration 
+file set up in step 1, using the `--secrets-config`:
+
+```bash
+go run main.go secrets init --secrets-config <PATH>
```
Comment (Kourin1996) [inline docs/how-tos/howto-setup-hashicorp-vault.md:74]: This is correct one?

```shell
go run main.go secrets init --config <PATH>
```

**Review by lazartravica — CHANGES_REQUESTED** [review]
@zivkovicmilos please assess @Kourin1996's comments and we're good to go with merging the KMS integration.

File: docs/how-tos/howto-setup-hashicorp-vault.md (hunk 1 above)
Comment (zivkovicmilos) [reply docs/how-tos/howto-setup-hashicorp-vault.md]: Good catch!

Fixed 👍 

4dc3836

File: docs/how-tos/howto-setup-hashicorp-vault.md (hunk 2 above)
Comment (zivkovicmilos) [reply docs/how-tos/howto-setup-hashicorp-vault.md]: Yep, another good catch 💯 

Fixed 👍 

4dc3836

**Review by Kourin1996 — APPROVED** [review]
LGTM 👌

**Pull request merged**